include/toml++/impl/path.h

```diff
@@ -0,0 +1,411 @@
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& append(std::wstring_view);
+
+		/// \brief	Prepends elements to the beginning of the TOML path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& prepend(std::wstring_view);
+
+		/// \brief	Replaces the contents of the path object by a new path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		path& assign(std::wstring_view);
+
+		/// \brief Returns a wstring representing the path
+		TOML_EXPORTED_MEMBER_FUNCTION
+		std::wstring wstring() const;
+
+#endif // TOML_ENABLE_WINDOWS_COMPAT
+
+	};
+
+	inline std::ostream& operator<<(std::ostream& os, const toml::path& rhs)
+	{
+		os << rhs.string();
+		return os;
+	}
+
+	inline std::istream& operator>>(std::istream& is, toml::path& rhs)
+	{
+		std::string s;
+		is >> s;
+		rhs.assign(s);
+
+		return is;
+	}
+
+	inline path operator+(const toml::path& lhs, const toml::path& rhs)
+	{
+		toml::path result = { lhs };
+		result.append(rhs);
+
+		return result;
+	}
+
+	inline path operator+(const toml::path& lhs, std::string_view rhs)
+	{
+		return lhs + toml::path(rhs);
+	}
+
+	inline path operator+(const toml::path& lhs, const char* rhs)
+	{
+		return lhs + toml::path(rhs);
+	}
+
+	inline path operator+(std::string_view lhs, const toml::path& rhs)
+	{
+		toml::path result = { rhs };
+		result.prepend(lhs);
+
+		return result;
+	}
+
+	inline path operator+(const char* lhs, const toml::path& rhs)
+	{
+		toml::path result = { rhs };
+		result.prepend(lhs);
+
+		return result;
+	}
+
+	inline path operator/(const toml::path& lhs, const toml::path& rhs)
+	{
+		return lhs + rhs;
+	}
+
+	inline path operator/(const toml::path& lhs, std::string_view rhs)
+	{
+		return lhs + rhs;
+	}
+
+	inline path operator/(const toml::path& lhs, const char* rhs)
+	{
+		return lhs + toml::path(rhs);
+	}
+
+	inline path operator/(std::string_view lhs, const toml::path& rhs)
+	{
+		return lhs + rhs;
+	}
+
+	inline path operator/(const char* lhs, const toml::path& rhs)
+	{
+		return lhs + rhs;
+	}
+
+#if TOML_ENABLE_WINDOWS_COMPAT
+
+	inline path operator+(const toml::path& lhs, std::wstring_view rhs)
+	{
+		return lhs + toml::path(rhs);
+	}
+
+	inline path operator+(const toml::path& lhs, const wchar_t* rhs)
+	{
+		return lhs + toml::path(rhs);
+	}
+
+	inline path operator+(std::wstring_view lhs, const toml::path& rhs)
+	{
+		toml::path result = { rhs };
+		result.prepend(lhs);
+
+		return result;
+	}
+
+	inline path operator+(const wchar_t* lhs, const toml::path& rhs)
+	{
+		toml::path result = { rhs };
+		result.prepend(lhs);
+
+		return result;
+	}
+
+	inline path operator/(const toml::path& lhs, std::wstring_view rhs)
+	{
+		return lhs + rhs;
+	}
+
+	inline path operator/(const toml::path& lhs, const wchar_t* rhs)
+	{
+		return lhs + toml::path(rhs);
+	}
+
+	inline path operator/(std::wstring_view lhs, const toml::path& rhs)
+	{
+		return lhs + rhs;
+	}
+
+	inline path operator/(const wchar_t* lhs, const toml::path& rhs)
+	{
+		return lhs + rhs;
+	}
+
+#endif // TOML_ENABLE_WINDOWS_COMPAT
+
```

Comment:
For non-member operators like this you almost always want to implement them in the class body and have them be `friend`; this makes them "[hidden friends](https://jacquesheunis.com/post/hidden-friend-compilation/)" which helps deal with ADL issues (+ potentially improve compile times in operator-overload-heavy scenarios)

Reply:
I had it set up this way initially but was having linker issues in test.cpp when trying to run the unit tests... unsure why, I'll move them back and have a closer look.